architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,261 @@
+
+###### Managing secrets for datastores
+
+Applications RP service has a Datastore RP. This RP is the resource provider for datstores such as SQL database, Mongo DB and Redis Cache. As of today, sensitive information such as DB connection string, user/ password that is required to provision these resources is stored in plain text. The feature which enables datastores to use a secret store is in progress. Until the feature is available, we recommend the users use recipes to deploy datastores. With recipe, the secrets are not stored in plain text.
+
+#### Access to cluster
+
+The RP can create and manage Kubernetes resources on behalf of the user. For example, it can deploy a container based on the image provided by the user. This container can execute arbitrary code and may create additional resources within the cluster where Radius is running or even in another cluster. The Application RP's service account allows the RP to securely authenticate with Kubernetes to perform these actions. The service account is configured with appropriate permissions to enable this. 
+
+#### Exposing User Application to Internet
+
+The Applications RP can create ingress Kubernetes objects. While deploying a Radius 'Gateway' resource, 
+the RP creates HTTPProxy objects, which expose Kubernetes service to outside of cluster. 
+
+#### Bicep Recipe execution
+
+Bicep recipes can provision arbitrary infrastructure resources in the cloud, and are provided by users. In order to execute a Bicep recipe, Applications RP's Recipe Engine first authenticates with a OCI compliant registry and then  downloads the recipe from it. RP uses the stored Azure credentials to authenticate with the registry. RP also supports federated credentials. When enabled, RP uses this to authenticate with the registry. Whenever possible, federated identies should be preferred and used for robust security. After retrieving the Bicep recipe, the RP requests UCP to deploy it.
+
+#### Terraform Recipe execution
+
+Terraform recipes are download from internet too. We support public and private modules as well as different kinds of authentication.
+
+Terraform downloads providers from the internet onto an empty directory `/terraform` which is mounted into the applications RP pod and executes them. This means that these providers have access to the network, filesystem, and environment variables, as well as the running memory of Applications RP. Terraform communicates with AWS as needed to deploy resources.
+
+#### Data Serialization / Formats
+
+We use custom parsers to parse Radius-related resource IDs and do not use any other custom parsers. The custom parser that parses Radius resource IDs has its own security mechanisms that don't accept anything other than a Radius resource ID. 
```

Comment:
> The custom parser that parses Radius resource IDs has its own security mechanisms that don't accept anything other than a Radius resource ID.

Does it?

I think this could be simplified to say something like.

> Radius resource ids are a custom string format that can be parsed from untrusted data. The parser is a shared component.

Beyond that is there anything else that needs to be said?